src/xdg-desktop-portal.c

```diff
@@ -431,13 +431,13 @@ on_name_lost (GDBusConnection *connection,
               const gchar     *name,
               gpointer         user_data)
 {
-  g_main_loop_quit (loop);
+  exit_with_status (0);
```

Comment:
I find exiting the "right" way (exiting the loop and returning from `main`) to be generally better, and allows for more proper cleanup.

Why are you changing to a more sudden exit without proper cleanup?

Reply:
It does call `g_main_loop_quit` and sets the status code, which IMO is better than just exiting without being explicit about the status code.

Reply:
Ah, exit_with_status is an internal helper, I was grepping in the wrong tree..